scripts/reborn_webui_v2_live_qa/run_live_qa.py

```
@@ -3739,45 +3948,93 @@ async def _wait_for_slack_delivery_marker(
                         "slack_history": history,
                         "approval_attempts": approval_attempts[-5:],
                     }
-                if isinstance(outcome, dict) and outcome.get("outcome") not in (None, "delivered"):
+                notice_state = _classify_triggered_notice_outcome(outcome)
+                if notice_state == "notifier_failure":
                     raise AssertionError(
-                        "triggered Slack delivery completed without delivered outcome: "
+                        "background-run notifier recorded a terminal failure "
+                        "for the fire (notices could not be delivered): "
                         f"run={row!r} outcome={outcome!r} history={history!r}"
                     )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)